lib/components/PlayerScreen/queue_list.dart

```diff
@@ -1054,34 +1053,70 @@ class QueueSectionHeader extends StatelessWidget {
   Widget build(context) {
     final queueService = GetIt.instance<QueueService>();
 
-    return StreamBuilder(
-      stream: Rx.combineLatest2(
-          queueService.getPlaybackOrderStream(),
-          queueService.getLoopModeStream(),
-          (a, b) => PlaybackBehaviorInfo(a, b)),
-      builder: (context, snapshot) {
-        PlaybackBehaviorInfo? info = snapshot.data;
-
-        return Padding(
-          padding: const EdgeInsets.only(left: 16.0, right: 16.0),
-          child: Row(
-            crossAxisAlignment: CrossAxisAlignment.center,
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: [
-              Expanded(
-                child: GestureDetector(
-                    child: Padding(
-                      padding: const EdgeInsets.symmetric(vertical: 8.0),
-                      child: title,
-                    ),
-                    onTap: () {
-                      if (source != null) {
-                        navigateToSource(context, source!);
-                      }
-                    }),
-              ),
-              if (controls)
-                Row(
+    return Padding(
+      padding: const EdgeInsets.only(left: 16.0, right: 16.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          Expanded(
+            child: GestureDetector(
+                child: Padding(
+                  padding: const EdgeInsets.only(bottom: 8.0, top: 12.5),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      title,
+                      StreamBuilder(
+                          stream: queueService.getQueueStream(),
+                          builder: (context, snapshot) {
+                            if (snapshot.hasData) {
+                              var trackPosition = snapshot
+                                      .data!.previousTracks.length +
+                                  (snapshot.data!.currentTrack == null ? 0 : 1);
+                              var trackCount = trackPosition +
+                                  snapshot.data!.nextUp.length +
+                                  snapshot.data!.queue.length;
+                              var remaining = const Duration(seconds: 0);
+                              for (var item in snapshot.data!.queue +
+                                  snapshot.data!.nextUp) {
+                                remaining += item.item.duration ??
+                                    const Duration(seconds: 0);
+                              }
```

Comment:
Could you please make these four things (current index, total queue size, remaining queue length, remaining duration, maybe also total duration) getters on `FinampQueueInfo`? That would clean up the code, and it might be useful to have in other places too.